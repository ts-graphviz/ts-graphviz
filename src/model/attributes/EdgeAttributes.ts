import { edgeAttributesSet } from '../values/util/attributesSets';
import { ArrowheadLike, ColorLike, LabelLike } from '../values/util/likeTypes';
import { Attributes } from './Attributes';

/**
 * @category Attributes
 */
export class EdgeAttributes extends Attributes {
  public set(key: 'label', value: LabelLike): void;
  public set(key: 'color', value: ColorLike): void;
  public set(key: 'arrowhead', value: ArrowheadLike): void;
  // public set(key: string, value: any): void;
  public set(key: string, value: any): void {
    if (edgeAttributesSet.has(key)) {
      super.set(key, value);
    } else {
      throw new Error('Not implemented.');
    }
  }
}
