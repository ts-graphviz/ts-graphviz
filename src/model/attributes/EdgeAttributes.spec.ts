import { EdgeAttributes } from './EdgeAttributes';

describe('class EdgeAttributes', () => {
  let attrs: EdgeAttributes;
  beforeEach(() => {
    attrs = new EdgeAttributes();
  });

  describe('renders correctly by toDot method', () => {
    it('no attributes', () => {
      expect(attrs.toDot()).toMatchSnapshot();
    });

    it('one attribute', () => {
      attrs.set('label', 'test');
      expect(attrs.toDot()).toMatchSnapshot();
    });

    it('some attributes', () => {
      attrs.set('label', 'test');
      attrs.set('color', 'red');
      expect(attrs.toDot()).toMatchSnapshot();
    });
  });
});
