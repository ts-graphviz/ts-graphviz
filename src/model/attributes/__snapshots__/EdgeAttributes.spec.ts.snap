// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class EdgeAttributes renders correctly by toDot method no attributes 1`] = `""`;

exports[`class EdgeAttributes renders correctly by toDot method one attribute 1`] = `
"[
  label=\\"test\\",
]"
`;

exports[`class EdgeAttributes renders correctly by toDot method some attributes 1`] = `
"[
  label=\\"test\\",
  color=\\"red\\",
]"
`;
